Zend/zend_objects.c

```diff
@@ -253,9 +262,17 @@ ZEND_API void ZEND_FASTCALL zend_objects_clone_members(zend_object *new_object,
 		} ZEND_HASH_FOREACH_END();
 	}
 
-	if (old_object->ce->clone) {
+	if (has_clone_method) {
+		zval *prop;
+
 		GC_ADDREF(new_object);
 		zend_call_known_instance_method_with_0_params(new_object->ce->clone, new_object, NULL);
+
+		rebuild_object_properties(new_object);
```

Comment:
This should not be necessary, you can just do something like:

```c
for (uint32_t i = 0; i < ce->default_properties_count; i++) {
	zval *prop_info = ce->properties_info_table[i];
	zval *zv = OBJ_PROP_NUM(new_object, i);
	...
}
```

Reply:
Nice, thanks! :)